src/HttpHeader.h

```diff
@@ -194,6 +196,14 @@ int httpHeaderParseQuotedString(const char *start, const int len, String *val);
 SBuf httpHeaderQuoteString(const char *raw);
 
 void httpHeaderCalcMask(HttpHeaderMask * mask, Http::HdrType http_hdr_type_enums[], size_t count);
+
+inline bool
+HttpHeader::chunked() const
+{
+    return has(Http::HdrType::TRANSFER_ENCODING) &&
+           hasListMember(Http::HdrType::TRANSFER_ENCODING, "chunked", ',');
```

Comment:
AFAICT, this method does not need its second component anymore. Please move its defitinion to the declaration and just return `has(Http::HdrType::TRANSFER_ENCODING)`.

Reply:
> [Amos](https://github.com/squid-cache/squid/pull/702#discussion_r464808252): that caller may set a non-chunked coding

There are no such callers today, and I see no plausible scenario where they would be allowed in the future. If you want to protect us from (more) future bugs, you can (by adjusting putStr() implementation and adding a new method dedicated to setting Transfer-Encoding), but I do not think those protections are _necessary_. And even if they are deemed desirable, it is OK to leave them outside this "surgical" PR IMO because the problem is unlikely to manifest itself and does not affect Squid performance or functionality. Those hypothetical non-trivial changes are only about protecting future us from rather unlikely coding mistakes.

Reply:
My understanding of your argument against using a flag was that there might become such users on grounds that the API to set arbitrary headers exists. The corollary to that (theoretical) situation is the ability to set non-chunked codings into the header.

And yes there are such callers today. The httpHdrMangleList() logic is one.  Leaving that second argument in place covers what I expect to be the most common situations without going so far as to increase scope (and bonus: it shrinks the patch).

Reply:
> My understanding of your argument against using a flag was that there might become such users on grounds that the API to set arbitrary headers exists

My argument against the initial (incorrect) flag-based implementation attempt was that existing Squid code makes that flag stale. No future/hypothetical/admin-dependent problems. The suggested code was simply broken.

The same problem does not exist for a has()-based implementation. For a has()-based implementation to become broken in a similar fashion, we would have to add new code (and new functionality) that does not look like anything we have today. That is why I am comfortable with not addressing that future (unlikely) problem in this PR. Please note that I am not against addressing that problem in this PR either -- I do not recommend doing it here for the reasons already [stated](https://github.com/squid-cache/squid/pull/702#discussion_r465207738), but I would not object to (properly) addressing it if your risk assessment is different.

> And yes there are such callers today. The httpHdrMangleList() logic is one.

I do not see how httpHdrMangleList() would break a has()-based implementation. If somebody misconfigures Squid to add custom Transfer-Encoding headers, then we are not responsible for the outcome, _and_ the Squid code would "work" as badly with the `chunked` member check as without it (if it makes sense to compare sets of broken use cases at all -- I doubt it does).

Reply:
Fine. Done in 2fc34818a